Bring up Android and Browser cross-compilers on Windows

directhex opened the pull request:
Thus far, only Android x64 cross compiler - we have been very lax with separating our use of HOST_WINDOWS vs TARGET_WINDOWS, and attempting to cross compile for other architectures fails horribly. This should get us at least partial progress towards the goal of platform parity on Windows with Linux.

jkoritzinsky committed:
Start creating runtime, apphost, and targeting packs using the new SDK in the NewSFX subset.

jkoritzinsky committed:
Fix generation of runtime package, runtime symbols package, and host symbols package.

jkoritzinsky committed:
Correcly include hosting files and DIA in the shared framework archives and packages

jkoritzinsky committed:
Include PackageOverrides.txt in the ref pack.

jkoritzinsky committed:
Fix product brand prefix.

jkoritzinsky committed:
Add the shared host installer.

jkoritzinsky committed:
Add the hostfxr build and clean up MacOS building.

jkoritzinsky committed:
Rename host and hostfxr projects to ensure we don't collide with the nuget packaging projects.

jkoritzinsky committed:
Add bundle installer.

jkoritzinsky committed:
Merge branch 'master' of https://github.com/dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Update platform manifest entries to include linux/mac file variations from the runtime.

jkoritzinsky committed:
Update apphost pack to match master.

jkoritzinsky committed:
Fix typo

jkoritzinsky committed:
Include manpages and fix linux package properties.

jkoritzinsky committed:
Fix linux package properties.

jkoritzinsky committed:
Match short-description from packages in master.

jkoritzinsky committed:
Update to uploaded package.

jkoritzinsky committed:
Update metadata and SDK to generate an exact match of the productbuild xml file.

jkoritzinsky committed:
First draft moving crossgen2 to new SDK.

jkoritzinsky committed:
Get crossgen2 pack working with updated SDK.

jkoritzinsky committed:
Merge remote-tracking branch 'dotnet/master' into adopt-unified-sharedfx

jkoritzinsky committed:
Build archives from the installer and bundle projects.

jkoritzinsky committed:
Move the dotnet-runtime-deps installers to use the new SDK.

jkoritzinsky committed:
Rough attempt to convert the build to use the new sfx sdk.

jkoritzinsky committed:
Remove now empty subsets.

jkoritzinsky committed:
Update installer subsets for testing.

jkoritzinsky committed:
Remove non-existent packaging project from subsets.

jkoritzinsky committed:
Add missing Windows api shim dll to platform manifest.

jkoritzinsky committed:
Update sharedfx sdk.

jkoritzinsky committed:
Add both versions of the long-name dac to the platform manifest.

jkoritzinsky committed:
Skip builds for non-relevant components on Mono.

jkoritzinsky committed:
Remove installer tests by default temporarily to validate product build.

jkoritzinsky committed:
Remove pkgprojs subset from mono installer build since it doesn't build the hosts.

jkoritzinsky committed:
Skip the apphost pack on mono.

jkoritzinsky committed:
Update host test infra to use the new sfx projects.

jkoritzinsky committed:
Delete an out of date test (the alternative case this test exercises is now the only case in this repo)

jkoritzinsky committed:
Fix packaging tests and remove NETStandard packaging tests since we aren't producing that ref pack any more.

jkoritzinsky committed:
Fixes for test official build failures. Re-eneable installer tests by default.

jkoritzinsky committed:
Fix condition for building linux installers.

jkoritzinsky committed:
Exclude Mono header files from the deps.json. They don't need to be in it.

jkoritzinsky committed:
Don't include the host files on the sfxproj.

jkoritzinsky committed:
Add mono-specific files to the platform manifest.

jkoritzinsky committed:
Remove extra copy step in installer job.

jkoritzinsky committed:
Include libmonosgen-2.0.a in platform manifest.

jkoritzinsky committed:
Fix change in linux-x64 test steps.

jkoritzinsky committed:
Fix display name of linux package steps.

jkoritzinsky committed:
Build the installers based on the portable build when applicable so they can find their artifacts.

jkoritzinsky committed:
The installer build no longer consumes Microsoft.NETCore.Platforms and Microsoft.NETCore.Targets, try removing the dependency on the AllConfigurations leg from the installer build.

jkoritzinsky committed:
Set PackageBrandNameSuffix for runtime-deps packages.

jkoritzinsky committed:
Updated sharedfx sdk

jkoritzinsky committed:
Fix debian runtime-deps dependency elements.

jkoritzinsky committed:
Merge branch 'master' of https://github.com/dotnet/runtime into adopt-unified-sharedfx-official-build-test

jkoritzinsky committed:
Add new platform manifest entries.

jkoritzinsky committed:
Add icudt.dat to the platform manifest.

jkoritzinsky committed:
Invert conditions to correctly skip generating installers on non-applicable platforms.

jkoritzinsky committed:
Merge branch 'adopt-unified-sharedfx-official-build-test' of /mnt/d/source/runtime into adopt-unified-sharedfx-official-build-test

jkoritzinsky committed:
Create extra installer subset so we can build just the installers (and avoid double-building nuget packages). Rename the newsfx subset and folder to sfx.

jkoritzinsky committed:
Fix Crossgen2 package name.

jkoritzinsky committed:
Merge branch 'master' into adopt-unified-sharedfx

jkoritzinsky committed:
Clean up some dead MSBuild goo and update the sharedfx package so we can override the crossgen2 package name without breaking the default package id logic.

jkoritzinsky committed:
Update SharedFX SDK.

jkoritzinsky committed:
Merge branch 'master' into adopt-unified-sharedfx

jkoritzinsky committed:
Fix discovery of the built dotnet layout for installer tests. Clean up some MSBuild along this path.

jkoritzinsky committed:
Update to use the new refactored packages.

jkoritzinsky committed:
Fix macOS build failures.

jkoritzinsky committed:
Fix linux installer build.

jkoritzinsky committed:
Upgrade to newest version of the SDK and centrally manage versions.

jkoritzinsky committed:
Merge branch 'master' of https://github.com/dotnet/runtime into adopt-unified-sharedfx-official-build-test

jkoritzinsky committed:
Update SDK to fix build skipping for the shared framework sdk.

jkoritzinsky committed:
Update SDK with full fix for skipbuild with packaging.

jkoritzinsky committed:
Another try for fixing the build-skip infra.

jkoritzinsky committed:
Upgrade SDK.

jkoritzinsky committed:
Fix package ids for mono packages.

jkoritzinsky committed:
Fix installer tests after update.

jkoritzinsky committed:
Add System.Private.Runtime.InteropServices.Javascript.dll to platform manifest

jkoritzinsky committed:
Merge branch 'master' of https://github.com/dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Fix Crossgen2 package to place all files in the tools folder.

jkoritzinsky committed:
Update to SDK with validation targets.

jkoritzinsky committed:
Add new ICU files to manifest.

jkoritzinsky committed:
Crossgen2 doesn't need to go through framework resolution so we can permit non-file-versioned dlls. (jitinterface.dll is currently unversioned).

Signed-off-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

jkoritzinsky committed:
Put the property in a property group.

jkoritzinsky committed:
Merge branch 'master' of https://github.com/dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Ensure we build corehost before sfx since we no longer directly build the project from the sfx projects.

jkoritzinsky committed:
Add missing entry for additional ICU data file.

jkoritzinsky committed:
Remove step downloading all artifacts and remove MSBuild glue to hook up looking up artifacts in the full download.

Signed-off-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

jkoritzinsky committed:
Merge branch 'master' into adopt-unified-sharedfx

jkoritzinsky committed:
Update SDK

Signed-off-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

jkoritzinsky committed:
Update SDK to get MSI fix.

Signed-off-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

jkoritzinsky committed:
Merge branch 'master' into adopt-unified-sharedfx

jkoritzinsky committed:
Don't even try building the hosting, crossgen2, or bundle installers when RuntimeFlavor is Mono.

jkoritzinsky committed:
Remove out of date comment.

Signed-off-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

jkoritzinsky committed:
Merge branch 'master' of github.com:dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Rename subsets based on PR discussion.

jkoritzinsky committed:
Update platform manifest and fix path to 3rd party notices for installer.

jkoritzinsky committed:
Remove test feed.

jkoritzinsky committed:
Update subset references I missed.

jkoritzinsky committed:
Merge branch 'master' of github.com:dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Fixed the subset arguments in other yaml files.

jkoritzinsky committed:
Merge branch 'master' of github.com:dotnet/runtime into adopt-unified-sharedfx

jkoritzinsky committed:
Merge branch 'master' of github.com:dotnet/runtime into adopt-unified-sharedfx

directhex committed:
Merge remote-tracking branch 'origin/master' into android-cross2

directhex committed:
Redesign mono.aotcross building

directhex committed:
Rejiggle so AOT compiler gets copied to HOST/cross/TARGET dir

directhex committed:
Merge remote-tracking branch 'origin/master' into android-cross2

directhex committed:
Merge remote-tracking branch 'origin/master' into android-cross2

directhex committed:
Build a package with the cross compiler inside (hardcoded to android-x64 thus far)

directhex committed:
Parallelize, apparently successfully

directhex committed:
Pull in changes from the other PR

directhex committed:
Only build cross-compiler packs on Linux FOR NOW, until we have NDK on Win/Mac

directhex committed:
Fix building locally

directhex committed:
Fix up subset behaviour

directhex committed:
Build cross in pipeline, and enable cross packages on Mac

directhex committed:
Nope, build cross on osx_x64 job, not android_x64 job

directhex committed:
Release not debug?

directhex committed:
Fix old-style Android build

directhex committed:
Workaround for x86 builds

directhex committed:
Try something else

directhex committed:
Mumble mumble

directhex committed:
Gosh darn it

directhex committed:
Rename android offset file to help disambiguate

directhex committed:
Create a new property which can generate offsets only

directhex committed:
Revert "Nope, build cross on osx_x64 job, not android_x64 job"

This reverts commit 2364432941eac0ffa9f2193868205c59cded79d7.

directhex committed:
Revert "Nope, build cross on osx_x64 job, not android_x64 job"

This reverts commit 2364432941eac0ffa9f2193868205c59cded79d7.

directhex committed:
Clean up more pipeline errors

directhex committed:
Add new yaml branch for generating offset files

directhex committed:
Merge remote-tracking branch 'origin/master' into android-cross

directhex committed:
Fix sysroot setting

directhex committed:
Missing escaping

directhex committed:
Why can't PublishPipelineArtifact do wildcards>?!

directhex committed:
Fic typos

directhex committed:
See if dependencies work

directhex committed:
Ensure Mono jobs pass through dependsOn

directhex committed:
Try adding AotCross to build. This should fail as we don't consume offsets yet

directhex committed:
Try downloading offsets

directhex committed:
Don;t hide behind conditional

directhex committed:
Fix offset aertifact name

directhex committed:
Download to correct place

directhex committed:
Copy offset file where it needs to be

directhex committed:
debug

directhex committed:
add debug, fix path

directhex committed:
typo

directhex committed:
Fix escaping

directhex committed:
Fix building cross compilers with no NDK, but an offset file

directhex committed:
Get rid of debug noise

directhex committed:
Merge remote-tracking branch 'origin/master' into android-cross

directhex committed:
Fix noise on Windows

directhex committed:
Build cross on correct platforms

directhex committed:
Build all android targets

directhex committed:
Try and work around llvm-config not being found on windows

directhex committed:
Revert "Get rid of debug noise"

This reverts commit 2f9a661e7d214a2fd6e91a5eefc89f2adec43097.

directhex committed:
More debug spew, osx should be working

directhex committed:
tpyo

directhex committed:
tweaky

directhex committed:
Get rid of a debug output

Commit:
Android cross WIP on windows

akoeplinger committed:
Fix issue with llvm libs

directhex committed:
Merge remote-tracking branch 'origin/master' into steveisok/android-cross

directhex committed:
Pass CPPFLAGS to AOT configure on Windows

directhex committed:
Typo

directhex committed:
More fixups

directhex committed:
Fix up condition

directhex committed:
Midpoint of progress, should get over 300 objects built

directhex committed:
_builds_ AOT, doesn't link yet

directhex committed:
One linker issue fixed, 2 to go

directhex committed:
Fix linking

directhex committed:
Windows likes ".exe"

directhex committed:
typo

directhex committed:
.exe!!!

directhex committed:
Get mono-threads-coop building in cross

directhex committed:
no interp for AOT runtime

directhex committed:
Make mini-llvm build w/ wasm

directhex committed:
Make Windows/Android cross only do x64 for now

directhex committed:
Add Windows cross-compilers to CI

directhex commented:
Ignore the number of commits, merging from master is always fun.

directhex committed:
Force use of Ninja

directhex committed:
Use Ninja for pack call too, or build breaks

directhex committed:
Handle Windows compiler init better

directhex committed:
Revert "Handle Windows compiler init better"

This reverts commit 37be82d361d95b4d9bf8a1d6b869d75374dca9f5.

directhex committed:
Deal with vcvarsall AOT?

directhex commented:
Failure appears to be because we use the 32-bit->64-bit cross compiler. On our 64-bit builds of 64-bit Mono for 64-bit Windows. SOMEWHERE IN THE UNIVERSE THIS IS THE RIGHT WAY TO DO COMPUTERS

directhex committed:
Ensure cmake cache written to correct folder

directhex committed:
Since i can't repro locally, dump build file from CI

directhex committed:
typo

directhex committed:
sigh

directhex committed:
try fixing cmake bool syntax

directhex committed:
WHY WERE THESE CHANGES DROPPED

directhex committed:
Revert "WHY WERE THESE CHANGES DROPPED"

This reverts commit 8a4d8705ef5486e696a9c6213ab89b5ad8ace00e.

directhex committed:
Revert "try fixing cmake bool syntax"

This reverts commit a09efb312210db7508f7bba52dce8c1bf127986c.

directhex committed:
Capture cmake version, to help with repro

directhex committed:
and ninja

directhex committed:
Maybe the secret lies in cmakecache.txt

directhex committed:
Add a trim

directhex committed:
Revert "Add a trim"

This reverts commit 334b46fb5f1b4bd11ec53a2a91da38408f1fab63.

directhex committed:
Don't use NormalizeDirectory, it breaks on Windows

pranavkm commented on src/mono/mono.proj:
```diff
@@ -396,6 +397,38 @@
     </PropertyGroup>
 
+    <!-- Windows specific options -->
+    <ItemGroup Condition="'$(OS)' == 'Windows_NT'">
```
```suggestion
    <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
```

directhex committed:
Update src/mono/mono.proj

Co-authored-by: Pranav K <prkrishn@hotmail.com>

directhex committed:
Merge branch 'master' into android-cross

directhex committed:
fix define check

directhex committed:
Use init-compiler for AOT build on windows, not just AOT configure

Apparently, CMake on Windows doesn't codify INCLUDE/LIB into the build system from configure, just CC/CCX, so if you run build without initializing the compiler again, you get failures from missing .h files

directhex committed:
Escape quotes?

directhex committed:
trim llvm dir ending

directhex committed:
Get rid of debug spew

directhex committed:
Merge remote-tracking branch 'origin/master' into steveisok/android-cross

lewing commented:
Failure looks like https://github.com/dotnet/runtime/pull/48075

lewing committed:
Merge remote-tracking branch 'upstream/master' into android-cross

steveisok commented:
Looks good. Way to swim through the pathing issues 👍
